rtos/Thread.cpp

```diff
@@ -265,11 +268,7 @@ uint32_t Thread::max_stack() {
     _mutex.lock();
 
     if (_tid != NULL) {
-        os_thread_t *thread = (os_thread_t *)_tid;
-        uint32_t high_mark = 0;
-        while (((uint32_t *)(thread->stack_mem))[high_mark] == 0xE25A2EA5)
-            high_mark++;
-        size = thread->stack_size - (high_mark * sizeof(uint32_t));
+        size = osThreadGetStackSize(_tid) - osThreadGetStackSpace(_tid);
```

Comment:
Ah, I forgot there were two watermarking schemes. So the existing code would actually work when OS_STACK_WATERMARK isn't set. Either way works for me. Defining OS_STACK_WATERMARK seems like it might unify the code more, so that would be my preference.